File: projects/m1/017-what-day-of-the-week/solutions/valentinaBazzacchi/solution-what-day-of-the-week.js
let year = prompt("Insert a year");

let dayOfWeek;    
dayOfWeek = (+year + (Math.floor((year - 1) / 4)) - Math.floor((year - 1) / 100) + Math.floor((year - 1) / 400)) % 7;
    
switch (dayOfWeek) {
    case 0:
    dayOfWeek = "Sunday";
    break;
    case 1:
    dayOfWeek = "Monday";
    break;
    case 2:
    dayOfWeek = "Tuesday";
    break;
    case 3:
    dayOfWeek = "Wednesday";
    break;
    case 4:
    dayOfWeek = "Thursday";
    break;
    case 5:
    dayOfWeek = "Friday";
    break;
    case 6:
    dayOfWeek = "Saturday";
    }
    
    alert("The day of the week for January 1 of the year " + year + " is " + dayOfWeek);

File: projects/m1/017-what-day-of-the-week/solutions/jimmyhu/solution-what-day-of-the-week.py
year = int(input('enter the year: '));
import math
day_of_the_week = (year + math.floor((year - 1) / 4) - math.floor((year - 1) / 100) + math.floor((year - 1) / 400)) % 7;
days = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday']
print(days[day_of_the_week])